src/main/java/de/tum/in/www1/artemis/service/compass/umlmodel/classdiagram/UMLAttribute.java

```diff
@@ -61,13 +61,32 @@ public double similarity(Similarity<UMLElement> reference) {
 
         UMLAttribute referenceAttribute = (UMLAttribute) reference;
 
+        if (!parentsSimilar(referenceAttribute)) {
+            return similarity;
+        }
+
         similarity += NameSimilarity.levenshteinSimilarity(name, referenceAttribute.getName()) * CompassConfiguration.ATTRIBUTE_NAME_WEIGHT;
 
         similarity += NameSimilarity.nameEqualsSimilarity(attributeType, referenceAttribute.getAttributeType()) * CompassConfiguration.ATTRIBUTE_TYPE_WEIGHT;
 
         return ensureSimilarityRange(similarity);
     }
 
+    /**
+     * Checks if the parent classes of this attribute and the given reference attribute are similar/equal by comparing the similarity IDs of both parent classes. If the similarity
+     * IDs are not set, it calculates the similarity of the parent classes itself and checks against the configured equality threshold.
+     *
+     * @param referenceAttribute the reference attribute of which the parent class is compared against the parent class of this attribute
+     * @return true if the parent classes are similar/equal, false otherwise
+     */
+    private boolean parentsSimilar(UMLAttribute referenceAttribute) {
```

Comment:
Same as in `UMLMethod.java`

Reply:
Also, this method is duplicated. Would it make sense to have an other hierarchy level here, such as `UMLElementWithParent`?

Reply:
I thought about it, but decided not to introduce an additional hierarchy level. It would increase the complexity of the hierarchy without adding much value. I think it is okay to have this small method in two places in this case.

However, I changed the check for the similarityId to be more explicit as in `UMLMethod.java`.